### PR TITLE
Give gufunc a name

### PR DESCRIPTION
**Pull request by leofang**
Currently if we throw `_GUFunc` to `cupyx.time.repeat` we get this error:
```
AttributeError: '_GUFunc' object has no attribute '__name__'
```
So this PR fixes it. Though, I am not sure if assigning the name of the wrapped function to the `_GUFunc` object is a good idea. Suggestions welcome!

### CODE REVIEW

File: cupy/_core/_gufuncs.py [hunk 1]
```diff
@@ -177,6 +177,7 @@ def __init__(self, func, signature, **kwargs):
         self._func = func
         self._signature = signature
         self._default_casting = 'same_kind'
+        self.__name__ = func.__name__
```
Comment (emcastillo) [inline cupy/_core/_gufuncs.py:180]: How about
```
self.__name__ = kwargs.get('name', func.__name__)
```
😄

Comment (leofang) [reply cupy/_core/_gufuncs.py:180]: Not bad!

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
Jenkins, test this please

**Review by emcastillo — APPROVED** [review]

**Pull request merged**